### PR TITLE
Add closing Browser Tab on iOS

### LINKED ISSUE #4: How to close browser tab?
It looks that there is no way to close the browser tab. I think this is required to implement the OAuth flow.

### PR DESCRIPTION
**Pull request by chuchuva**
Fixes #4

### CODE REVIEW

File: plugin/src/ios/CBTBrowserTab.h [hunk 1]
```diff
@@ -18,8 +18,14 @@
 #import <SafariServices/SafariServices.h>
 
 @interface CBTBrowserTab : CDVPlugin
+{
```
Comment (WilliamDenniss) [inline plugin/src/ios/CBTBrowserTab.h]: move opening bracket to previous line

File: plugin/src/ios/CBTBrowserTab.h [hunk 2]
```diff
@@ -18,8 +18,14 @@
...
+    @protected
+    SFSafariViewController* _safari;
+}
 
 - (void) isAvailable:(CDVInvokedUrlCommand*)command;
 - (void) openUrl:(CDVInvokedUrlCommand*)command;
+- (void) close:(CDVInvokedUrlCommand*)command;
```
Comment (WilliamDenniss) [inline plugin/src/ios/CBTBrowserTab.h]: Add a space before the asterisk. `- (void) close:(CDVInvokedUrlCommand *)command;` (can you update the two previous lines as well?).

File: plugin/src/ios/CBTBrowserTab.m [hunk 3]
```diff
@@ -63,4 +62,11 @@ - (void) openUrl:(CDVInvokedUrlCommand*)command {
                                 callbackId:command.callbackId];
 }
 
+- (void) close:(CDVInvokedUrlCommand*)command {
+    if (!_safari)
```
Comment (WilliamDenniss) [inline plugin/src/ios/CBTBrowserTab.m]: please add brackets, i.e. `if (!_safari) {`…

File: plugin/src/ios/CBTBrowserTab.h [hunk 4]
```diff
@@ -18,8 +18,14 @@
 #import <SafariServices/SafariServices.h>
 
 @interface CBTBrowserTab : CDVPlugin
+{
+    @protected
+    SFSafariViewController* _safari;
```
Comment (WilliamDenniss) [inline plugin/src/ios/CBTBrowserTab.h]: can we call it` _safariViewController` so it's clear it's not mobile safari?

File: plugin/src/ios/CBTBrowserTab.m [hunk 5]
```diff
@@ -50,10 +50,9 @@ - (void) openUrl:(CDVInvokedUrlCommand*)command {
                                     callbackId:command.callbackId];
     }
 
-    SFSafariViewController *sfvc =
-            [[SFSafariViewController alloc] initWithURL: url];
+    _safari = [[SFSafariViewController alloc] initWithURL:  url];
```
Comment (WilliamDenniss) [inline plugin/src/ios/CBTBrowserTab.m]: remove space before `url`, i.e. `initWithURL:url]`

File: plugin/www/browsertab.js [hunk 6]
```diff
@@ -23,3 +23,9 @@ exports.openUrl = function(url, opt_error) {
   var error = (!opt_error) ? doNothing : opt_error;
     exec(doNothing, error, 'BrowserTab', 'openUrl', [url]);
 };
+
+exports.close = function(opt_error) {
+  var doNothing = function() {};
+  var error = (!opt_error) ? doNothing : opt_error;
+    exec(doNothing, error, 'BrowserTab', 'close', []);
```
Comment (WilliamDenniss) [inline plugin/www/browsertab.js]: this line should be 2-space indented, not 4

**Comment by chuchuva** [comment]
@WilliamDenniss Done! Please review again.

**Comment by iainmcgin** [comment]
I still don't know if this is the right approach, because a close() method of this form cannot work for Android. @WilliamDenniss do we want platform-specific methods in this library?

**Comment by WilliamDenniss** [comment]
Thanks for your change, it was merged as part of #9.

**Pull request closed without merging**